mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+        List of all valid methods.
+
+    classes_: NDArray
+        Array with the name of each class.
+
+    n_classes_: int
+        Number of classes that are in the training dataset.
+
+    uncalib_pred: NDArray
+        Array of the uncalibrated predictions.
+
+    calibrators: Dict[str, RegressorMixin]
+        Dictionnary of all the fitted calibrators.
+
+    References
+    ----------
+    [1] Gupta, Chirag, and Aaditya K. Ramdas. "Top-label calibration
+    and multiclass-to-binary reductions." arXiv preprint
+    arXiv:2107.08353 (2021).
+
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from mapie.calibration import MapieCalibrator
+    >>> X_toy = np.arange(9).reshape(-1, 1)
+    >>> y_toy = np.stack([0, 0, 1, 0, 1, 2, 1, 2, 2])
+    >>> mapie = MapieCalibrator().fit(X_toy, y_toy, random_state=20)
+    >>> y_calib = mapie.predict_proba(X_toy)
+    >>> print(y_calib)
+    [[0.84900723 0.         0.        ]
+     [0.75432411 0.         0.        ]
+     [0.62285341 0.         0.        ]
+     [0.         0.33333333 0.        ]
+     [0.         0.33333333 0.        ]
+     [0.         0.33333333 0.        ]
+     [0.         0.         0.33333002]
+     [0.         0.         0.54326683]
+     [0.         0.         0.66666124]]
+    """
+
+    fit_attributes = [
+        "estimator",
+        "calibrators",
+    ]
+
+    valid_calibrators = {
+        "sigmoid": _SigmoidCalibration(),
+        "isotonic": IsotonicRegression(out_of_bounds="clip")
+    }
+
+    valid_methods = ["top_label"]
+
+    def __init__(
+        self,
+        estimator: Optional[ClassifierMixin] = None,
+        method: str = "top_label",
+        calibrator: Optional[Union[str, RegressorMixin]] = None,
+        cv: Optional[str] = "split",
+    ) -> None:
+        self.estimator = estimator
+        self.method = method
+        self.calibrator = calibrator
+        self.cv = cv
+
+    def _fit_calibrator(
+        self,
+        item: int,
+        calibrator: RegressorMixin,
+        y_calib: ArrayLike,
+        top_class_prob: NDArray,
+        y_pred: NDArray,
+        sample_weight: Optional[ArrayLike],
+    ) -> RegressorMixin:
+        """
+        Fitting the calibrator requires that we have a binary setting in place,
+        hence, we find the subset of values on which we want to apply this
+        calibration and apply a binary calibration on these.
+
+        Parameters
+        ----------
+        item : int
+            The class for which we fit a calibrator.
+        calibrator : RegressorMixin
+            Calibrator to fit.
+        y_calib : NDArray of shape (n_samples,)
+            Training labels.
+        top_class_prob : NDArray of shape (n_samples,)
+            The independent values of the calibrator, it represents the
+            maximum score of the predictions.
+        y_pred : NDArray of shape (n_samples,)
+            The array to determine which class the maximum prediction belongs
```

Comment:
Why not just "Array of predictions" ?

Reply:
or Predictions, as you write "Sample weights"

Reply:
done.